src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.spec.ts

```diff
@@ -61,12 +64,67 @@ describe('OsdRecvSpeedModalComponent', () => {
         default: 0
       }
     ];
+    spyOn(configurationService, 'filter').and.returnValue(observableOf(configOptions));
   });
 
   it('should create', () => {
     expect(component).toBeTruthy();
   });
 
+  describe('ngOnInit', () => {
+    /* I don't know why this test case leads to an error in the following (no
```

Comment:
Hi, looked into the tests, it only fails if unit testing not in `DEV` mode, that said it seems that angular is not resetting it's stubs. A workaround is to `callThrough` the spies.

I've done that and added a `beforeEach` to initiate all spies there and call `ngOnInit`.
After that I removed some tests as I think they are not needed in order to reduce the coupling between code and tests.

For example, you don't have to spy if `getCurrentValues` got called, as the result oriented tests (:+1:) would fail instantly, as the object they need weren't there.

So now the diff :)

```diff
diff --git a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.spec.ts b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.spec.ts
index fb6bd729a1..b373629132 100755
--- a/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.spec.ts
+++ b/src/pybind/mgr/dashboard/frontend/src/app/ceph/cluster/osd/osd-recv-speed-modal/osd-recv-speed-modal.component.spec.ts
@@ -72,36 +72,16 @@ describe('OsdRecvSpeedModalComponent', () => {
   });
 
   describe('ngOnInit', () => {
-    /* I don't know why this test case leads to an error in the following (no
-    matter which) test case.
-    If you have an idea please uncomment the test and fix it, thanks ;)
-    it('should call getCurrentValues', () => {
-      const getCurrentValues = spyOn(component, 'getCurrentValues');
-      component.ngOnInit();
-      expect(getCurrentValues).toHaveBeenCalled();
-    });*/
-
-    it('should call detectPriority', () => {
-      const detectPriority = spyOn(component, 'detectPriority');
-      component.ngOnInit();
-      expect(detectPriority).toHaveBeenCalled();
-    });
-
-    it('should call setPriority', () => {
-      const setPriority = spyOn(component, 'setPriority');
-      component.ngOnInit();
-      expect(setPriority).toHaveBeenCalled();
-    });
+    let setPriority;
+    let setValidators;
 
-    it('should call setDescription', () => {
-      const setDescription = spyOn(component, 'setDescription');
+    beforeEach(() => {
+      setPriority = spyOn(component, 'setPriority').and.callThrough();
+      setValidators = spyOn(component, 'setValidators').and.callThrough();
       component.ngOnInit();
-      expect(setDescription).toHaveBeenCalled();
     });
 
     it('should call setValidators', () => {
-      const setValidators = spyOn(component, 'setValidators');
-      component.ngOnInit();
       expect(setValidators).toHaveBeenCalled();
     });
 
@@ -109,13 +89,10 @@ describe('OsdRecvSpeedModalComponent', () => {
       const defaultPriority = _.find(component.priorities, (p) => {
         return _.isEqual(p.name, 'default');
       });
-      const setPriority = spyOn(component, 'setPriority');
-      component.ngOnInit();
       expect(setPriority).toHaveBeenCalledWith(defaultPriority);
     });
 
     it('should set descriptions correctly', () => {
-      component.ngOnInit();
       expect(component.priorityAttrs['osd_max_backfills'].desc).toBe('');
       expect(component.priorityAttrs['osd_recovery_max_active'].desc).toBe('');
       expect(component.priorityAttrs['osd_recovery_max_single_start'].desc).toBe('');
```

Reply:
Cool, thanks for taking a look. I will update the tests accordingly.